### PR TITLE
[fix](sequence) value predicates shouldn't be push down when has sequence column

### LINKED ISSUE #20409: [Bug] wrong result using value predicate when has sequence column
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Version

master and branch-1.2-lts

### What's Wrong?

CREATE TABLE `test_uniq_value_predicate` (
  `k` int(11) NULL,
  `int_value` int(11) NULL,
  `char_value` char(10) NULL,
  `date_value` date NULL
) ENGINE=OLAP
UNIQUE KEY(`k`)
COMMENT 'OLAP'
DISTRIBUTED BY HASH(`k`) BUCKETS 1
PROPERTIES (
"replication_allocation" = "tag.location.default: 1",
"in_memory" = "false",
"storage_format" = "V2",
"function_column.sequence_col" = "int_value",
"disable_auto_compaction" = "false"
);

insert into test_uniq_value_predicate values(0, 2, 'test int', '2000-02-02');
insert into test_uniq_value_predicate values(0, 1, 'test char', '2000-01-01');

select * from test_uniq_value_predicate;
+------+-----------+------------+------------+
| k    | int_value | char_value | date_value |
+------+-----------+------------+------------+
|    0 |         2 | test int   | 2000-02-02 |
+------+-----------+------------+------------+

 select * from test_uniq_value_predicate where k=0 && int_value=2;
+------+-----------+------------+------------+
| k    | int_value | char_value | date_value |
+------+-----------+------------+------------+
|    0 |         2 | test int   | 2000-02-02 |
+------+-----------+------------+------------+

select * from test_uniq_value_predicate where k=0 && int_value=1;
+------+-----------+------------+------------+
| k    | int_value | char_value | date_value |
+------+-----------+------------+------------+
|    0 |         1 | test char  | 2000-01-01 |
+------+-----------+------------+------------+

### What You Expected?

empty result when int_value=1

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by liaoxin01**
## Proposed changes

Issue Number: close #20409

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by liaoxin01** [comment]
run buildall

File: be/src/olap/rowset/beta_rowset_reader.cpp [hunk 1]
```diff
@@ -304,10 +304,12 @@ Status BetaRowsetReader::next_block_view(vectorized::BlockView* block_view) {
 
 bool BetaRowsetReader::_should_push_down_value_predicates() const {
     // if unique table with rowset [0-x] or [0-1] [2-y] [...],
-    // value column predicates can be pushdown on rowset [0-x] or [2-y], [2-y] must be compaction and not overlapping
+    // value column predicates can be pushdown on rowset [0-x] or [2-y], [2-y]
+    // must be compaction, not overlapping and don't have sequence column
     return _rowset->keys_type() == UNIQUE_KEYS &&
```
Comment (yiguolei) [inline be/src/olap/rowset/beta_rowset_reader.cpp:309]: If _should_push_down_value_predicates returns false, then the value predicates is not evaluated in segment iterator. Then when are these value predicates evaluated？

**Comment by liaoxin01** [comment]
run buildall

File: be/src/olap/rowset/beta_rowset_reader.cpp (hunk 1 above)
Comment (liaoxin01) [reply be/src/olap/rowset/beta_rowset_reader.cpp:309]: These value predicates will be evaluated by scan node.

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**